### PR TITLE
Update diff bakend node tests for auto update doc

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

- Extend `onnx/backend/test/cmd_tools.py` with a new option: "diff", which will only update the backend node test data whose script has been updated.
- Make `auto_update_doc.yml` CI utilize "--diff" and update the backend node test data accordingly.

### Motivation and Context

https://github.com/onnx/onnx/pull/5450 "auto update doc" was introduced for helping users update docs like Operators.md and TestCoverage.md. It will be also useful that the CI can help users update backend node test as well.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Potentially we should just update everything in CI because the CI machines are the most stable, compared to machines developers use?

**Review by xadupre — APPROVED** [review]

**Comment by jcwchen** [comment]
> Potentially we should just update everything in CI because the CI machines are the most stable, compared to machines developers use?

We have discussed this a lot in our offline meetings, but I think there are some concerns to make the update only happen in the CI. Let me conclude:
- Operators.md is useful for PR reviewers by showing in the PR
- Providing backend node test in the PR is useful for the PR author to verify them in their local

Also we have another option to remove Operators.md entirely: https://github.com/onnx/onnx/issues/5571. For now, I think making "auto update doc" update docs/backend test data is a fine temporary solution. PR authors can decide to update them from their own, or simply reply on the CI to update for them. Going forward we can discuss further and see whether we can eventually make these updates all happen in the CI.

File: onnx/backend/test/case/node/__init__.py [hunk 1]
```diff
@@ -21,8 +23,10 @@
     TypeProto,
 )
 
+
```
Comment (justinchuby) [inline onnx/backend/test/case/node/__init__.py:26]:
```suggestion
```

**Comment by jcwchen** [comment]
> I meant we didn’t have to find the —diff and update. We can update all node tests (when ppl choose to use the auto gen label)

In that way some of irrelevant files (input.pb or output.pb) might be updated unintentionally, because NumPy behaves differently on different platforms. The CI is running on Linux and for those old data created from Windows will be changed in the CI. I think having "--diff" makes more sense to let the generation tool only update those corresponding test data.

**Comment by gramalingam** [comment]
Two points. (i) Extending the utility with the --diff option is a good thing, it will help even devs who want to do it locally for their own validation. (ii) Whether the CI should use that option or not is orthogonal. I am fine with using the --diff. (Ideally it should make no difference.)

**Review by gramalingam — APPROVED** [review]

**Pull request merged**